proposals/TP1-xds-transport-next.md

```diff
@@ -0,0 +1,864 @@
+    provide full delta diffs, since there is no explicit collection object to
+    inform the client when resources come and go.
+
+### Transport network addressing
+
+An `xdstp://` URI specifies a logic authority and it’s necessary to map to some physical 
+network address during the resource fetch process. In the existing xDS configuration
+fetch configuration, this capability is provided by a `ConfigSource` delivered alongside a
+resource name in the xDS resource graph.
+
+This proposal adds to `ConfigSource` the ability to specify the
+authorities for which the `ConfigSource` may be used:
+
+```
+message Authority {
+  string name = 1;
+
+  … when signing is added, items such as CA trust chain, cert pinning … 
+}
+
+message ConfigSource {
+  repeated Authority authorities = N;
+
+  …  extant ConfigSource … 
+}
+```
+
+When the `ConfigSource` adjacent to a resource name matches the URI authority,
+the `ConfigSource` will continue to be used.
+
+When there is no match, the xDS client will fallback to the bootstrap to map
+from authority to `ConfigSource`.
+
+
+To support ADS with multiple control planes, the
+[`ApiType`](https://github.com/envoyproxy/envoy/blob/31128e7dc22355876020188bc8feb99304663041/api/envoy/config/core/v3/config_source.proto#L44)
+enum in `ApiConfigSource` will be augmented with `AGGREGATED_DELTA_GRPC`.
+This will replace and deprecate the existing `ads` field in `ConfigSource`. When
+used in `xdstp://` authority resolution, an `ConfigSource` will act as an ADS
+transport if configured with this new enum value. Instead of pointing to a
+single shared ADS `ConfigSource` declared in the bootstrap, any resource
+authority (regardless of type) that maps to a `ConfigSource` with
+`AGGREGATED_DELTA_GRPC` will be multiplexed on the xDS stream specified by the
+`ConfigSource`.
+
+### Discovery request and responses
+
+State-of-the-world (SotW) `Discovery{Request,Response}` are not supported with
+the new naming scheme and are effectively deprecated for servers that make use
+of the scheme. The use of inlined entries in list collections allows simple
+management servers that do not wish to perform state tracking to perform
+state-of-the-world updates. This is achieved by sending the entire state, e.g.
+every listener inlined in a `ListenerCollection`.
+
+In both the xDS transport and resources, existing fields for managing resource
+names will continue to be used. A client and server will detect `xdstp://`
+semantics by examining the prefix of a resource string.
+
+Clients are expected to have knowledge ahead of time (via mechanisms not part of
+this proposal, e.g. support documentation) on whether the servers specified in
+the bootstrap supports the new URI conventions.
+
+### Server redirects
+
+The server may issue redirects by including an `xds::core::v3::ResourceLocator`
+in a `Resource` wrapped in the `Any` message. Redirects are only supported for
```

Comment:
Does it still make sense to use `xds.core.v3.ResourceLocator` on the wire for redirects, since we're no longer using that proto message for anything else on the wire?  I'm wondering if we should instead just have something simple like this:

```
message Redirect {
  string resource_locator = 1;
}
```

Then a redirect would be triggered by having the `Resource` include a message of type `xds.core.v3.Redirect`.